torch_geometric/loader/neighbor_loader.py

```diff
@@ -13,12 +13,14 @@
     to_csc,
     to_hetero_csc,
 )
-from torch_geometric.typing import EdgeType, InputNodes
-
-NumNeighbors = Union[List[int], Dict[EdgeType, List[int]]]
+from torch_geometric.typing import InputNodes, NodeType, NumNeighbors
+from torch_geometric.utils import mask_to_index
 
 
 class NeighborSampler:
+
+    device = "cpu"
```

Comment:
I'm really not a fan of doing this, but it seemed a bit pointless to make it configurable, and I think its better than hard coding it in more places

Reply:
What's the reason for this change?

Reply:
I did this initially because I was storing something on CPU in the LinkNeighbourSampler too, so it seemed better to make sure they all shared the same device.. I'll revert